.github/workflows/swift-toolchain.yml

```diff
@@ -5091,26 +5097,109 @@ jobs:
               -p:ToolchainVariants="`"asserts;noasserts`"" `
               ${{ github.workspace }}/SourceCache/swift-installer-scripts/platforms/Windows/bundle/installer.wixproj
 
-      - if: ${{ inputs.release }}
+      - name: Generate Build Provenance (offline installer)
+        if: ${{ inputs.release }}
         uses: actions/attest-build-provenance@v2
         with:
           subject-path: ${{ github.workspace }}/BinaryCache/installer/Release/${{ matrix.arch }}/installer.exe
 
-      - uses: actions/upload-artifact@v4
+      - name: Upload installer (offline)
+        uses: actions/upload-artifact@v4
+        with:
+          name: Windows-${{ matrix.arch }}-installer-offline
+          path: ${{ github.workspace }}/BinaryCache/installer/Release/${{ matrix.arch }}/installer.exe
+    
+      - name: Generate online installer download URL
+        if: ${{ inputs.release }}
+        id: generate-base-release-download-prefix
+        run: |
+          $BaseReleaseDownloadUrlPrefix = "${{ inputs.swift_tag }}/${{ matrix.arch }}/${{ github.run_id }}"
+          echo "url_prefix=$BaseReleaseDownloadUrlPrefix" >> $env:GITHUB_OUTPUT
```

Comment:
Minor: without https:// it's not really a prefix but rather a slug (url subcomponent).